msg-socket/src/req/socket.rs

```diff
@@ -74,19 +82,20 @@ impl<T: ClientTransport> ReqSocket<T> {
             .map_err(|e| ReqError::Transport(Box::new(e)))?;
 
         // Create the socket backend
+        let mut pending_requests = FxHashMap::default();
+        pending_requests.reserve(self.options.max_active_requests);
         let driver = ReqDriver {
             options: Arc::clone(&self.options),
             id_counter: 0,
             from_socket,
             conn: Framed::new(stream, reqrep::Codec::new()),
-            egress_queue: VecDeque::new(),
-            // TODO: we should limit the amount of active outgoing requests, and that should be the capacity.
-            // If we do this, we'll never have to re-allocate.
-            pending_requests: FxHashMap::default(),
+            egress_queue: VecDeque::with_capacity(self.options.max_active_requests),
+            pending_requests,
             socket_state: Arc::clone(&self.state),
             timeout_check_interval: tokio::time::interval(Duration::from_millis(
```

Comment:
Missed this last time but we can replace this with `timeout_check_interval: tokio::time::interval(self.options.timeout / 10)`